cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/docker_reg_v2_repo_lister.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021-2022 the Kubeapps contributors.
+// SPDX-License-Identifier: Apache-2.0
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	log "k8s.io/klog/v2"
+
+	"helm.sh/helm/v3/pkg/registry"
+
+	// ORAS => OCI Registry AS Storage
+	orasregistryv2 "oras.land/oras-go/v2/registry"
+	orasregistryremotev2 "oras.land/oras-go/v2/registry/remote"
+	orasregistryauthv2 "oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// This flavor of OCI lister Works with respect to those OCI registry vendors that implement
+// Docker Registry API V2 or OCI Distribution Specification. For example, GitHub (ghcr.io)
+// References:
+// - https://docs.docker.com/registry/spec/api/#base
+// - https://github.com/opencontainers/distribution-spec/blob/main/spec.md#api
+func NewDockerRegistryApiV2RepositoryLister() OCIRepositoryLister {
+	return &dockerRegistryApiV2RepositoryLister{}
+}
+
+type dockerRegistryApiV2RepositoryLister struct {
+}
+
+// ref https://github.com/distribution/distribution/blob/main/docs/spec/api.md#api-version-check
+// also https://github.com/oras-project/oras-go/blob/14422086e418/registry/remote/registry.go
+func (l *dockerRegistryApiV2RepositoryLister) IsApplicableFor(ociRegistry *OCIRegistry) (bool, error) {
+	log.Infof("+IsApplicableFor(%s)", ociRegistry.url.String())
+
+	orasRegistry, err := newRemoteOrasRegistry(ociRegistry)
+	if err != nil {
+		return false, err
+	} else {
+		ping := "OK"
+		err = orasRegistry.Ping(context.Background())
+		if err != nil {
+			ping = fmt.Sprintf("%v", err)
+		}
+		log.Infof("ORAS Registry [%s] Ping: %s", ociRegistry.url.String(), ping)
+		return err == nil, err
+	}
+}
+
+// ref https://github.com/distribution/distribution/blob/main/docs/spec/api.md#listing-repositories
+func (l *dockerRegistryApiV2RepositoryLister) ListRepositoryNames(ociRegistry *OCIRegistry) ([]string, error) {
+	log.Infof("+ListRepositoryNames()")
+
+	orasRegistry, err := newRemoteOrasRegistry(ociRegistry)
+	if err != nil {
+		return nil, err
+	} else {
+		// this is the way to stop the loop in
+		// https://github.com/oras-project/oras-go/blob/14422086e41897a44cb706726e687d39dc728805/registry/remote/registry.go#L112
+		done := errors.New("(done) backstop")
```

Comment:
see above. Please ignore that part

Reply:
Have +1'd on that advice. Personally prefer to not include WIP stuff in a PR that is landing, and instead just have stubs (as you had already with the hard-coded value), just so new functionality is added as a unit in context, rather than piecemeal WIP parts over time. But if you find it easier this way, that's fine :)

Reply:
thanks Michael